core/field_variable.js

```diff
@@ -68,23 +77,40 @@ Blockly.FieldVariable.prototype.init = function() {
   this.initModel();
 };
 
+/**
+ * Initialize the model for this field if it has not already been initialized.
+ * If the value has not been set to a variable by the first render, we make up a
+ * variable rather than let the value be invalid.
+ * @package
+ */
 Blockly.FieldVariable.prototype.initModel = function() {
-  if (!this.getValue()) {
-    // Variables without names get uniquely named for this workspace.
-    var workspace =
-        this.sourceBlock_.isInFlyout ?
-            this.sourceBlock_.workspace.targetWorkspace :
-            this.sourceBlock_.workspace;
-    this.setValue(Blockly.Variables.generateUniqueName(workspace));
-  }
-  // If the selected variable doesn't exist yet, create it.
-  // For instance, some blocks in the toolbox have variable dropdowns filled
-  // in by default.
-  if (!this.sourceBlock_.isInFlyout) {
-    this.sourceBlock_.workspace.createVariable(this.getValue());
+  if (this.variable_) {
+    return; // Initialization already happened.
+  }
+  this.workspace_ = this.sourceBlock_.workspace;
```

Comment:
This is slightly different than before and I don't understand why. Mostly I don't understand why the workspace.targetWorkspace is a thing we have at all, so I can't justify to myself why it is okay for it not to be there anymore :)

Reply:
Mostly this just moved into getOrCreateVariable, and the concept of potential variables has replaced whatever we were doing for flyout variables before.  

A flyout workspace has a target workspace; the main workspace doesn't.  In a flyout workspace we don't want to create a real variable if we don't have to, because it wouldn't go away when the flyout closes.  Previously we were just coming up with a unique name and calling setValue, but not actually creating a variable model.  Now it's required to act on a variable model, so we create one.